### PR TITLE
feat: implement habit editing functionality

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR
Added habit editing functionality and renamed the color field to colorName for consistency

https://github.com/user-attachments/assets/9ad807f9-cd4e-4c6c-8f8c-0eb2d2625209

### What changed?
- Renamed `color` field to `colorName` in habit creation schema
- Added new `useEditHabit` hook for updating existing habits
- Enhanced the edit-habit screen to support both creation and editing modes
- Added edit functionality to the habit card's menu
- Updated form defaults to populate with existing habit data when editing

### How to test?
1. Create a new habit to verify creation still works
2. Click the menu (three dots) on any habit card
3. Select "Edit habit"
4. Modify any fields (title, description, color, icon, or multiple completions)
5. Click "Save Changes"
6. Verify the habit updates with the new information
7. Verify the success toast appears

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/147?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#147** <a href="https://app.graphite.dev/github/pr/habits-together/app/147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/147?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#146** <a href="https://app.graphite.dev/github/pr/habits-together/app/146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#145** <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#144** <a href="https://app.graphite.dev/github/pr/habits-together/app/144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]
